extensions/resource-deployment/src/test/services/notebookService.test.ts

```diff
@@ -0,0 +1,294 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata';
+import 'mocha';
+import * as path from 'path';
+import * as should from 'should';
+import * as sinon from 'sinon';
+import * as TypeMoq from 'typemoq';
+import * as vscode from 'vscode';
+import { NotebookPathInfo } from '../../interfaces';
+import { Notebook, NotebookService } from '../../services/notebookService';
+import { IPlatformService 	} from '../../services/platformService';
+import * as loc from '../../localizedConstants';
+import assert = require('assert');
+
+class Deferred<T> {
```

Comment:
Put this in common utils class (can just be in test folder for now since extension doesn't need it)